### ISSUE TITLE
Is there a way to find the probability or confidence score of the extracted named entities

### ISSUE THREAD

**Issue by Nitin54** [RasaHQ/rasa_nlu#676]
**Rasa NLU version** (e.g. `0.7.3`):

**Used backend / pipeline** (`mitie`, `spacy_sklearn`, ...):

**Operating system** (windows, osx, ...):

**Issue**:

**Content of configuration file** (if used & relevant):
```json

```

**Comment by wrathagom** [comment]
Good link, but haven't gone deep enough to figure out what it would take inside of Rasa, but my guess is that it would be easier if first implemented in spacy.

https://github.com/explosion/spaCy/issues/831

**Comment by tmbo** [comment]
This is something we might be able to easily provide for the CRF (although I'd call them scores not probabilities) - to support the spacy build in ones there need to be changes to spacy (not sure if there is something like that in spacy 2.0)

**Comment by honnibal** [comment]
Current spaCy 2 recipe for this:

```python

# Number of alternate analyses to consider. More is slower, and not necessarily better -- you need to experiment on your problem.
beam_width = 16
# This clips solutions at each step. We multiply the score of the top-ranked action by this value, and use the result as a threshold. This prevents the parser from exploring options that look very unlikely, saving a bit of efficiency. Accuracy may also improve, because we've trained on greedy objective.
beam_density = 0.0001 
nlp = spacy.load('en_core_web_sm')

docs = list(nlp.pipe(texts, disable=['ner']))
beams = nlp.entity.beam_parse(docs, beam_width=beam_width, beam_density=beam_density)

for doc, beam in zip(docs, beams):
    entity_scores = defaultdict(float)
    for score, ents in nlp.entity.get_beam_parses(beam):
        for start, end, label in ents:
            entity_scores[(start, end, label)] += score
```

This functionality is important for [Prodigy](https://prodi.gy), so we'll definitely continue to support probabilities. The function above is currently considered "internals" though, because the API isn't so nice. So you might have to watch for breaking changes on this.

**Comment by tmbo** [comment]
Ok that sounds doable, thanks @honnibal 

@honnibal Any news on a timeline for the RC? Would love to get started supporting it

**Comment by honnibal** [comment]
https://github.com/explosion/spaCy/releases/tag/v2.0.0

We got to 2.0.0a19 and it was stable, so we decided there wasn't much point having an extended "release candidate" phase. There are far fewer bugs than v1, so it made more sense to just publish it :)

We added back the `nlp.path` attribute btw, which was the first problem you guys were hitting. Training, saving and loading is now really solid too: https://spacy.io/usage/training

**Comment by sharukh2308** [comment]
"This is something we might be able to easily provide for the CRF (although I'd call them scores not probabilities) - to support the spacy build in ones there need to be changes to spacy (not sure if there is something like that in spacy 2.0)"

- Is it possible to test this functionality now using the ner_crf component?

**Comment by wrathagom** [comment]
This too has been merged at least for CRF #883

**Comment by folterj** [comment]
As this seems to be the only current source of scoring, can the above code be updated so it works with the current version of spacy please?
